packages/material-ui/src/Tabs/Tabs.test.js

```diff
@@ -383,241 +366,248 @@ describe('<Tabs />', () => {
+        Object.defineProperty(tablist, 'clientWidth', { value: 120 });
+        Object.defineProperty(tablist, 'scrollWidth', { value: 216 });
+        tablist.scrollLeft = 96;
 
-        assert.strictEqual(hasLeftScrollButton(wrapper), true);
-        assert.strictEqual(hasRightScrollButton(wrapper), false);
+        setProps();
+        expect(hasLeftScrollButton(container)).to.equal(true);
+        expect(hasRightScrollButton(container)).to.equal(false);
       });
 
       it('should set only right scroll button state', () => {
-        const wrapper = mount(tabs);
+        const { container, setProps, getByRole } = render(
+          <Tabs
+            value={0}
+            variant="scrollable"
+            scrollButtons="on"
+            ScrollButtonComponent={AccessibleTabScrollButton}
+            style={{ width: 200 }}
+          >
+            <Tab />
+            <Tab />
+            <Tab />
+          </Tabs>,
+        );
+        const tablist = getByRole('tablist');
 
-        setFakeTabs(wrapper, {
-          scrollLeft: 0,
-          scrollWidth: 110,
-          clientWidth: 100,
-        });
+        Object.defineProperty(tablist, 'clientWidth', { value: 120 });
+        Object.defineProperty(tablist, 'scrollWidth', { value: 216 });
+        tablist.scrollLeft = 0;
 
-        assert.strictEqual(hasLeftScrollButton(wrapper), false);
-        assert.strictEqual(hasRightScrollButton(wrapper), true);
+        setProps();
+        expect(hasLeftScrollButton(container)).to.equal(false);
+        expect(hasRightScrollButton(container)).to.equal(true);
       });
 
       it('should set both left and right scroll button state', () => {
-        const wrapper = mount(tabs);
-
-        setFakeTabs(wrapper, {
-          scrollLeft: 2,
-          scrollWidth: 110,
-          clientWidth: 100,
-        });
-
-        assert.strictEqual(hasLeftScrollButton(wrapper), true);
-        assert.strictEqual(hasRightScrollButton(wrapper), true);
-      });
-
-      it('should not set scroll button states if difference is only one pixel', () => {
-        const wrapper = mount(tabs);
+        const { container, setProps, getByRole } = render(
+          <Tabs
+            value={0}
+            variant="scrollable"
+            scrollButtons="on"
+            ScrollButtonComponent={AccessibleTabScrollButton}
+            style={{ width: 200 }}
+          >
+            <Tab />
+            <Tab />
+          </Tabs>,
+        );
+        const tablist = getByRole('tablist');
 
-        setFakeTabs(wrapper, {
-          scrollLeft: 0,
-          scrollWidth: 101,
-          clientWidth: 100,
-        });
+        Object.defineProperty(tablist, 'clientWidth', { value: 120 });
+        Object.defineProperty(tablist, 'scrollWidth', { value: 216 });
+        tablist.scrollLeft = 5;
 
-        assert.strictEqual(hasLeftScrollButton(wrapper), false);
-        assert.strictEqual(hasRightScrollButton(wrapper), false);
+        setProps();
```

Comment:
This is a noop. Used the same name as enzyme but a more accurate name would be `updateProps`. You don't set the props to `{}` you merge `{}` into the existing.

Reply:
The advantage of `setProps` in this case is to re-render the component. We don't need to provide new props.
Would you recommend a different approach? I can think of a different approach, we could provide an explicit method, for instance, we could override rerender to support zero arguments. Looking at the code, it doesn't seem to support this rerender use case outside of the box: https://github.com/testing-library/react-testing-library/blob/8e99f3cf99b76afbf2b20993745b802def3bb6c9/src/index.js#L67.

Reply:
> The advantage of setProps in this case is to re-render the component. 

Mainly because this doesn't happen in actual apps. Did you mean to simulate a scroll event?

Reply:
The test changes the underlying DOM layout measurements. The tabs component should check the measurements are each render to update its display accordingly.

Reply:
Yes. How would this happen in the browser?

Reply:
It will happen when the parent component rerender.